### PR TITLE
Update KMCP/PROFILE: add's nf-test, bumps versions, removes non-manadatory input channel

### PR DESCRIPTION
**Pull request by jfy133**
- Adds nf-test
- bumps-version
- Removes the actually non-mandatatory 'mode' value channel (to follow guidelines, but also needed for a bug fix in taxprofiler)

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/kmcp/profile/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    tag "kmcp/search"
+
+    setup {
+        run("UNTAR") {
+            script "../../../untar/main.nf"
+            process {
+                """
+                input[0] = [
+                                [id: 'test'],
+                                file(params.modules_testdata_base_path + 'delete_me/kmcp/kmcp_profile.tar.gz', checkIfExists: true)
```
Comment (SPPearce) [inline modules/nf-core/kmcp/profile/tests/main.nf.test:24]: Where is this file coming from? Do we need to put it somewhere better in test_data?

File: modules/nf-core/kmcp/profile/tests/nextflow.config
Comment (SPPearce) [inline modules/nf-core/kmcp/profile/tests/nextflow.config:1]: Do we need this config?

File: modules/nf-core/kmcp/profile/tests/main.nf.test (hunk 1 above)
Comment (jfy133) [reply modules/nf-core/kmcp/profile/tests/main.nf.test:24]: There isn't a better place unfortunately. It's only useable for this particular module/toolkit

**Review by Joon-Klaps — APPROVED** [review]
LGTM!

**Comment by jfy133** [comment]
> LGTM! 

Thanks!

**Pull request merged**